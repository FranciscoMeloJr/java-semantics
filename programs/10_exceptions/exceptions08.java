class main {

  int x;

  main(String[] args) {
    x = 5;
    try {
      if (true) throw new RuntimeException("a");
      System.out.print(x);       // should not print this
    } catch(RuntimeException y) {
      System.out.println(y.toString());  // should print this
    }
    System.out.println("Done!");
  }
}

public class exceptions08 {
  public static void main(String[] args) {
    new main(args);
  }
}

// 3
// Done!
