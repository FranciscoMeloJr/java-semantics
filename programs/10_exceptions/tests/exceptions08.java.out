java.lang.RuntimeException: a
Done!
